src/sse/sse_butterflies.rs

```diff
@@ -0,0 +1,4246 @@
+use core::arch::x86_64::*;
+use num_complex::Complex;
+
+use crate::{common::FftNum, FftDirection};
+
+use crate::array_utils;
+use crate::array_utils::{RawSlice, RawSliceMut};
+use crate::common::{fft_error_inplace, fft_error_outofplace};
+use crate::twiddles;
+use crate::{Direction, Fft, Length};
+
+use super::sse_common::{assert_f32, assert_f64};
+use super::sse_utils::*;
+
+#[allow(unused)]
+macro_rules! boilerplate_fft_sse_f32_butterfly {
+    ($struct_name:ident, $len:expr, $direction_fn:expr) => {
+        impl<T: FftNum> $struct_name<T> {
+            #[target_feature(enable = "sse3")]
```

Comment:
Did you do any comparisons between putting the `target_feature` here, vs on `perform_fft_contiguous` down below? Target_feature_ attributes prevent inlining, so if we put this here, that's 3 functions that can't be inlined, vs only one if we put it on `perform_fft_contiguous` -- which will end up increasing binary size.

Reply:
I have moved the `target_feature` around a bit and didn't measure any difference. The current state is where I left it after experimenting. Good point about inlining, I will clean it up.